lib/Net/SIP/SocketPool.pm

```diff
@@ -316,7 +317,8 @@ sub _timeout_sockets {
 	if ($tdiff>$expire) {
 	    $self->_del_socket($_);
 	} elsif ($_->{inside_connect} && $tdiff > $CONNECT_TIMEOUT) {
-	    $self->_del_socket($_,"connect timed out");
+	    my @cbs = $self->_del_socket($_,"connect timed out");
+	    invoke_callback($_) foreach @cbs;
```

Comment:
The callback returned from `_del_socket` via `delFD` is not necessarily the callback given to `sendto`.  I think it might be better to remember `$callback` from `sendto` inside the SocketPool object and call it from inside `_error` with the error description given there. This would also require no changes to EventLoop, thus maintaining the existing interface.

Reply:
I updated pull request. I'm not sure if this approach is correct, I stored callback inside `$fo` and called it directly with `ETIMEDOUT` errno from `_timeout_sockets`. Unit test is still working.